frappe/desk/doctype/calendar_view/calendar_view.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2017, Frappe Technologies and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Calendar View', {
+	onload: function(frm) {
+		frm.trigger('reference_doctype');
+	},
+	refresh: function(frm) {
+		if (!frm.is_new()) {
+			frm.add_custom_button(__('Show Calendar'),
+				() => frappe.set_route('List', frm.doc.reference_doctype, 'Calendar', frm.doc.name));
+		}
+	},
+	reference_doctype: function(frm) {
+		const { reference_doctype } = frm.doc;
+		if (!reference_doctype) return;
+
+		frappe.model.with_doctype(reference_doctype, () => {
+			const meta = frappe.get_meta(reference_doctype);
+
+			const subject_options = meta.fields.filter(
+				df => !frappe.model.no_value_type.includes(df.fieldtype)
+			).map(df => df.fieldname);
```

Comment:
`df.label` would be probably more appropriate for the select options

Reply:
yeah, but we want to store the fieldname in db and not label.

Reply:
Yeah, was thinking more along the lines of 'what gets shown and what gets stored'. However, if its already done in other places then not much point 👍